src/test/kotlin/org/rust/ide/inspections/lints/RsLivenessInspectionTest.kt

```diff
@@ -739,6 +739,23 @@ class RsLivenessInspectionTest : RsInspectionsTestBase(RsLivenessInspection::cla
         }
     """)
 
+    // https://github.com/intellij-rust/intellij-rust/issues/6513
+    fun `test remove function argument when function is used as an argument`() = checkFixByText("Remove parameter `x`", """
+        fn foo(<warning>x/*caret*/</warning>: i32) {}
+        fn bar<F>(<warning>f</warning>: F) where F: Fn((i32)) -> () {}
```

Comment:
Here refactoring would break code because `bar` expects function with one argument (probably would be nice to show some warning in such cases). Maybe it is better to use `id` here

```suggestion
        fn id<T>(t: T) { t }
```